### ISSUE TITLE
Incorrect documentation of `mapstruct.unmappedSourcePolicy` default value

### ISSUE THREAD

**Issue by cmcgowanprovidertrust** [mapstruct/mapstruct#3635]
I think there may be a mistake in the Reference Guide documentation. In [section 2.4](https://mapstruct.org/documentation/1.5/reference/html/#configuration-options) it describes the `mapstruct.unmappedSourcePolicy` and states it has a default value of `WARN`. But in practice, it seems that the default value is actually `IGNORE`. That is the behavior I get just setting up a new Mapper with no other configuration. 

The source code for `Mapper.java` seems to support the idea that `IGNORE` is the intended default: `ReportingPolicy unmappedSourcePolicy() default ReportingPolicy.IGNORE;`

I believe this was a simple copy-paste error introduced in https://github.com/mapstruct/mapstruct/pull/2572

**Comment by filiphr** [comment]
You are correct @cmcgowanprovidertrust. Would you be interested in providing a PR with that fix?

**Comment by cmcgowanprovidertrust** [comment]
Yes, just wanted to first confirm I was not missing something as my team just started using MapStruct. Filed https://github.com/mapstruct/mapstruct/pull/3637

**Comment by thunderhook** [comment]
Thanks alot @cmcgowanprovidertrust!

**Comment by cmcgowanprovidertrust** [comment]
Thank you all for the great library!
